R/plumber.R

```diff
@@ -673,6 +675,31 @@ plumber <- R6Class(
       }
 
       private$ends[[preempt]] <- c(private$ends[[preempt]], ep)
+    },
+    swaggerFileWalkMountsInternal = function(router, parentPath=""){
+
+      parentPath <- sub("[/]$", "", parentPath)
+      endpoints <- lapply(router$endpoints, function(endpoint){
+
+        endpointEntries <- lapply(endpoint, function(endpointEntry){
+          endpointEntry <- endpointEntry$clone()
+          endpointPath <- sub("^[/]", "", endpointEntry$path)
+          endpointPath <- paste(parentPath, endpointPath, sep="/")
+          endpointEntry$path <- endpointPath
+          endpointEntry
+        })
+        
+        endpointEntries
+      })
+
+      mounts <- router$mounts
+      mountedEndpoints <- unlist(lapply(names(mounts), function(mountPath){
+        mountedSubrouter <- router$mounts[[mountPath]]
+        mountPath <- sub("^[/]", "", mountPath)
+        mountedEndpoints <- private$swaggerFileWalkMountsInternal(mountedSubrouter, paste(parentPath, mountPath, sep="/"))
+      }))
+
+      c(endpoints, list(mountedEndpoints))
```

Comment:
Is there a reason you call `list` here? My understanding is that at this point `endpoints` is a list and `mountedEndpoints` is a character vector, so the result will definitely be a list.

Reply:
@alandipert I'm a bit fuzzy on the types but I think endpoints is a list of lists and mountedEndpoints without the extra `list` call is just a list. If you remove that call and try to run swagger documentation, you'll get `Error in for (e in fil) { : invalid for() loop sequence`. When I print out the result of calling `c` above without modification, I get entries like
```
[[1]]`$__no-preempt__14`
<PlumberEndpoint>
...
```

Reply:
Ah, that makes sense, thanks for clarifying.